lib/internal/crypto/keys.js

```diff
@@ -128,12 +130,33 @@ const [
   }
 
   const kAsymmetricKeyType = Symbol('kAsymmetricKeyType');
+  const kAsymmetricKeyDetails = Symbol('kAsymmetricKeyDetails');
+
+  // TODO: should the returne details object be frozen or otherwise protected
+  // from manipulation?
+  function normalizeKeyDetails(details = {}) {
+    if ('publicExponent' in details) {
```

Comment:
Checking `if (details.publicExponent !== undefined)` would be faster here ... the `if-in` syntax can be a bit slower.